metrics-prometheus/src/main/java/com/alibaba/metrics/prometheus/DubboMetricsExports.java

```diff
@@ -0,0 +1,187 @@
+        BucketCounter successCounter = compass.getBucketSuccessCount();
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        for (Map.Entry<String, BucketCounter> entry : compass.getErrorCodeCounts().entrySet()) {
+            String tag = entry.getKey();
+            long count = entry.getValue().getBucketCounts().get(start);
+            samples.add(sampleBuilder.createSample(metricName, "_bucket_count", Arrays.asList("category"), Arrays.asList(tag), count));
+        }
+        for (Map.Entry<String, BucketCounter> entry : compass.getAddonCounts().entrySet()) {
+            String tag = entry.getKey();
+            long count = entry.getValue().getBucketCounts().get(start);
+            samples.add(sampleBuilder.createSample(metricName, "_bucket_count", Arrays.asList("category"), Arrays.asList(tag), count));
+        }
+        long count = successCounter.getBucketCounts().get(start);
+        samples.add(sampleBuilder.createSample(metricName, "_bucket_count", Arrays.asList("category"), Arrays.asList("success"), count));
+        return new MetricFamilySamples(samples.get(0).name, Type.COUNTER, getHelpMessage(metricName.getKey(), compass), samples);
+    }
+
+    public MetricFamilySamples fromFastCompass(MetricName metricName, FastCompass fastCompass) {
+        long lastUpdateTime = fastCompass.lastUpdateTime();
+        int bucketInterval = fastCompass.getBucketInterval();
+        long start = getNormalizedStartTime(lastUpdateTime, bucketInterval);
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        Map<String, Map<Long, Long>> countPerCategory = fastCompass.getMethodCountPerCategory(start);
+        for (Map.Entry<String, Map<Long, Long>> entry: countPerCategory.entrySet()) {
+            if (entry.getValue().containsKey(start)) {
+                String tag = entry.getKey();
+                long count = entry.getValue().get(start);
+                samples.add(sampleBuilder.createSample(metricName, "bucket_count", Arrays.asList("category"), Arrays.asList(tag), count));
+            }
+        }
+        for (Map.Entry<String, Map<Long, Long>> entry: fastCompass.getMethodRtPerCategory(start).entrySet()) {
+            if (entry.getValue().containsKey(start)) {
+                String tag = entry.getKey();
+                long rt = entry.getValue().get(start);
+                samples.add(sampleBuilder.createSample(metricName, "bucket_sum", Arrays.asList("category"), Arrays.asList(tag), rt));
+            }
+        }
+        return new MetricFamilySamples(samples.get(0).name, Type.COUNTER, getHelpMessage(metricName.getKey(), fastCompass),
+                samples);
+    }
+
+    private MetricFamilySamples fromSnapshotAndCount(MetricName metricName, Snapshot snapshot, long count, double factor, String helpMessage) {
+        List<MetricFamilySamples.Sample> samples = Arrays.asList(
```

Comment:
For Timer, we still need .m1/.m5/.m15 and p50/p75/p95/p99/max/min/stddev/mean
For Histogram, we need p50/p75/p95/p99/max/min/stddev/mean